### PR TITLE
Add IList<IServiceDescriptor> property to IServiceCollection

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

**Comment by pranavkm** [comment]
Based on a discussion with @loudej - the intent is to expose the backing list to allow removing service descriptors that were previously added.

**Comment by davidfowl** [comment]
Why not just make IServiceCollection be an IList then?

**Comment by pranavkm** [comment]
You end up having to implement quite a few more properties to make the type be an IList. This is less hassle to create.

**Comment by pranavkm** [comment]
Updated.

File: src/Microsoft.Framework.DependencyInjection/ServiceCollection.cs [hunk 1]
```diff
@@ -14,6 +14,12 @@ public class ServiceCollection : IServiceCollection
         private readonly List<IServiceDescriptor> _descriptors = new List<IServiceDescriptor>();
 
         /// <inheritdoc />
+        public int Count => _descriptors.Count;
```
Comment (davidfowl) [inline src/Microsoft.Framework.DependencyInjection/ServiceCollection.cs]: Nice

**Comment by davidfowl** [comment]
I know @halter73 spoke about adding a .Replace method as well.

**Comment by pranavkm** [comment]
I'm actually using `Remove` to implement `Replace` in Mvc - I'll just move the code in here. The only issue is the `Replace` semantics isn't super clear - if you have more than service of the same type, is the Replace supposed to Replace all of them (i.e. RemoveAll + Insert) or just Replace the first one it encounters?

File: src/Microsoft.Framework.DependencyInjection/ServiceCollection.cs (hunk 1 above)
Comment (lodejard) [inline src/Microsoft.Framework.DependencyInjection/ServiceCollection.cs]: Wow

**Pull request closed without merging**